src/fields2cover/types/Path.cpp

```diff
@@ -114,6 +114,81 @@ std::string Path::serializePath(size_t digit_precision) const {
   return res;
 }
 
+/**
+ * @brief Discretize the swath sections of the path and return a new path
+ * @param step_size Discretization step in [m]
+ * @param swath_speed Robot swath speed in [m/s]
+ * @return New path with swath now discretized 
+*/
+Path Path::discretize_swath(double step_size, double swath_speed) const {
+    // Create new path
+    Path new_path;
+
+    // Loop through all the points in the path
+    for (size_t i = 0; i < this->size(); i++)
+    {
+        // Check if the current point is a SWATH point
+        if (this->states.at(i).type == f2c::types::PathSectionType::SWATH)
+        {
+            // We know that swaths are represented by only two point so the next point will
+            // be the end of the swath
+            // Save start and end swath points to variables
+            Point start_point = this->states.at(i).point;
+            Point end_point = this->states.at(i+1).point;
+
+            // We can now discretize the swath
+            // Calculate the distance between the start and end point
+            double distance = start_point.Distance(end_point);
+            // Calculate the number of points needed to discretize the swath
+            int num_points = floor(distance / step_size);
+
+            // Create vector's for the x, y and z values from start to end point
+            std::vector<double> x_values;
+            std::vector<double> y_values;
+            std::vector<double> z_values;
+            for (int j = 0; j < num_points; j++)
+            {
+                x_values.push_back(start_point.getX() + (j * (end_point.getX() - start_point.getX()) / num_points));
+                y_values.push_back(start_point.getY() + (j * (end_point.getY() - start_point.getY()) / num_points));
+                z_values.push_back(start_point.getZ() + (j * (end_point.getZ() - start_point.getZ()) / num_points));
+            }
+            // Add last point to list
+            x_values.push_back(end_point.getX());
+            y_values.push_back(end_point.getY());
+            z_values.push_back(end_point.getZ());
```

Comment:
if you make the previous loop instead of (for (int j = 0; j < num_points; j++)) --> (for (int j = 0; j <= num_points; j++)), these three lines are redundant.

Reply:
Fixed, Thank you!